BedrockServer.cpp

```diff
@@ -1656,30 +1640,37 @@ void BedrockServer::_reply(unique_ptr<BedrockCommand>& command) {
     }
 }
 
-void BedrockServer::suppressCommandPort(const string& reason, bool suppress, bool manualOverride) {
-    // If we've set the manual override flag, then we'll only actually make this change if we've specified it again.
-    if (_suppressCommandPortManualOverride && !manualOverride) {
-        return;
+
+void BedrockServer::blockCommandPort(const string& reason) {
+    lock_guard<mutex> lock(_portMutex);
+    _commandPortBlockReasons.insert(reason);
+    if (_commandPortBlockReasons.size() == 1) {
+        _commandPortPublic = nullptr;
+        _portPluginMap.clear();
     }
-    SINFO((suppress ? "Suppressing" : "Clearing") << " command port due to: " << reason);
+    SINFO("Blocking command port due to: " <<  reason << (_commandPortBlockReasons.size() > 1 ? " (already blocked)" : "") << ".");
+}
 
-    // Save the state of manual override. Note that it's set to *suppress* on purpose.
-    if (manualOverride) {
-        _suppressCommandPortManualOverride = suppress;
+void BedrockServer::unblockCommandPort(const string& reason) {
+    lock_guard<mutex> lock(_portMutex);
+    _commandPortLikelyBlocked = true;
+    auto it = _commandPortBlockReasons.find(reason);
+    if (it == _commandPortBlockReasons.end()) {
+        SWARN("Tried to unblock command port because: " << reason << ", but it wasn't blocked for that reason!");
```

Comment:
NAB might be helpful to indicate in this log line whether or not the port is currently blocked.

Reply:
It's a pain to serialize this, and it could be blocked for no reason or multiple reasons, and really, you're probably doing something wrong by trying to unblock it with whatever reason you sent, anyway, regardless of other reasons.

Reply:
I just meant whether or not is still blocked, not necessarily a serialization of the whole thing